### PR TITLE
[POC][IR] Initial stab at std::string->String upgrade

### PR DESCRIPTION
**Pull request by tqchen**
This PR is an example of String Update with backward compact(see json_compact.py)

### CODE REVIEW

**Comment by tqchen** [comment]
cc @jroesch @icemelon9

File: python/tvm/ir/json_compact.py [hunk 1]
```diff
@@ -84,12 +91,26 @@ def _update_global_key(item, _):
         del item["global_key"]
         return item
 
+    def _update_from_std_str(key):
+        def _convert(item, nodes):
+            str_val = item["attrs"][key]
+            jdata = json.loads(tvm.ir.save_json(tvm.runtime.String(str_val)))
```
Comment (zhiics) [inline python/tvm/ir/json_compact.py:97]: Should we bring the serialization of tvm:String out of save_json?

Comment (tqchen) [reply python/tvm/ir/json_compact.py:97]: Can you elaborate a bit about what do you mean? Right now all tvm objects are serialized through save_json, what we are doing here is to get that particular entry and append to the end of the nodes.

**Comment by tqchen** [comment]
@jroesch @zhiics 
wrt to the String serialization. Right now String is serialized as an normal object. So if the content of String is printable, it will show up as

```python
{type_key: "runtime.String", repr_str: "mystring"}
```

If the content of the string is not printable, then it will show up as 

```python
{type_key: "runtime.String", repr_b64: "base64encoding"}
```

So the new IR entry will look like (note that the name field actually stores the index to the node table
```
[
  {"type_key": "TypeVar", "attrs" : { "name": "2" }},
  {type_key: "runtime.String", repr_str: "myname"}
]
```
This mechanism is baked into json serialization so it is a bit hard to move out. Of course we could also say that String should mostly be printable, and use a special serialization(e.g. just print as str when it is an old  str attribute) Thet is a special case though and makes the json serialization harder to work with when directly serialize a String object itself.

**Comment by zhiics** [comment]
@tqchen ahh, I see. I was thinking if it is necessary to have similar stuff like Load/Save for tvm::String, so that we don't need to put them under serialization.cc. Looks it is not really needed. Thanks.

**Pull request merged**